ext/liquid_c/block.c

```diff
@@ -34,6 +35,9 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
     VALUE nodelist = rb_ivar_get(self, intern_nodelist);
 
     while (true) {
+        if (tokenizer->line_number != 0) {
```

Comment:
Oh, it's not a boolean, forget everything I said. This code is a bit confusing but makes sense, how about adding a comment somewhere saying that it uses `line_number` as a flag as well as a number?